### PR TITLE
Add image cleanup command and restructure folders

### PR DESCRIPTION
**Pull request by anothertobi**
The folder structure was very basic and it was complicated to isolate
single commands.

The image cleanup is based on
https://github.com/appuio/oc-plugins/blob/master/cleanup and reproduces
its behaviour except the delete prompt which is not intended by cobra.

### CODE REVIEW

File: kubernetes/basic.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	}
+
+	return
+}
+
+// Namespace from the kubeconfig
+func Namespace() (namespace string) {
+	namespace, _, err := kubeconfig().Namespace()
+	if err != nil {
+		panic(err)
```
Comment (ccremer) [inline kubernetes/basic.go]: Please avoid using panic(). We have a logging framework that can do the same (use the Fatal error level) and provide more meaningful errors for the user. You can also add fields to the logging message, similar to metadata in an ELK stack (e.g. what was the value of the namespace parameter)

Comment (ccremer) [reply kubernetes/basic.go]: e.g. `log.WithField("namespace", namespace).WithError(err).Fatal("Could not get namespace from Kubeconfig")` or whatever the reason for the fail might be.

Comment (anothertobi) [reply kubernetes/basic.go]: You are totally right, improved that on the whole codebase.

File: cmd/imagestream.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	cmd.Flags().BoolVarP(&o.Force, "force", "f", false, "delete image stream tags")
+	cmd.Flags().IntVarP(&o.CommitLimit, "git-commit-limit", "l", 100, "only look at the first <n> commits to compare with tags or use -1 for all commits")
+	cmd.Flags().StringVarP(&o.RepoPath, "git-repo-path", "p", ".", "absolute path to Git repository (for current dir use: $PWD)")
+	cmd.Flags().StringVarP(&o.Namespace, "namespace", "n", "", "Kubernetes namespace")
+	cmd.Flags().IntVarP(&o.Keep, "keep", "k", 10, "keep most current <n> images")
+	return cmd
+}
+
+func (o *ImageStreamCleanupOptions) cleanupImageStreamTags(cmd *cobra.Command, args []string) {
+	o.ImageStream = args[0]
```
[the remark this answers is not in the document]
Comment (anothertobi) [reply cmd/imagestream.go]: Added that 👍

File: pkg/cleanup/cleanup.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+// GetTagsMatchingPrefixes returns all tags matching one of the provided prefixes
+func GetTagsMatchingPrefixes(prefixes, tags *[]string) []string {
+	var matchingTags []string
+
+	if len(*prefixes) > 0 && len(*tags) > 0 {
+		for _, prefix := range *prefixes {
+			for _, tag := range *tags {
+				if strings.HasPrefix(tag, prefix) {
+					matchingTags = append(matchingTags, tag)
```
[the remark this answers is not in the document]
Comment (anothertobi) [reply pkg/cleanup/cleanup.go]: Added more debug logs

File: pkg/kubernetes/util_test.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package kubernetes
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var object = map[string]interface{}{"int": 20, "bool": true, "string": "foo"}
```
[the remark this answers is not in the document]
Comment (anothertobi) [reply pkg/kubernetes/util_test.go]: Added a Kubernetes object and moved to table driven tests.

**Pull request merged**